plugins/default/lib/disk_usage.py

```diff
@@ -1,35 +1,97 @@
+                        required=True, help="Warning Threshold")
+    parser.add_argument('-c', '--crit', type=int, action="store",
+                        required=True, help="Critical Threshold")
+    return parser
+
+
+def parse_df_line(line):
     mo = check_disk_output_pattern.match(line)
     if mo is None:
         return ()
-    return (mo.group(1), int(mo.group(2)), int(mo.group(3)))
+    return (mo.group(3), int(mo.group(1)), int(mo.group(2)))
+
+
+def parse_df_lines(lines):
+    return (filter(None, map(parse_df_line, lines.splitlines())))
+
+
+def analize(pod, disks, warning_threshold, critical_threshold):
+    results = []
+    for mount, space_usage, inode_usage in disks:
+        max_pcent = max(space_usage, inode_usage)
+
+        disk_status = nagios.UNKNOWN
+        if max_pcent >= critical_threshold:
+            disk_status = nagios.CRIT
+        elif max_pcent >= warning_threshold:
+            disk_status = nagios.WARN
+        elif max_pcent < warning_threshold:
+            disk_status = nagios.OK
+
+        results.append([pod, mount, space_usage, inode_usage, disk_status])
+    return results
+
 
-def analize(pcent, ipcent, warning_threshold, critical_threshold):
-    max_pcent = max(pcent, ipcent)
-    return max_pcent >= warning_threshold, max_pcent >= critical_threshold
+def report(results):
+    counts = Counter(d[4] for d in results)
+
+    ret = max(counts)
+
+    if ret == nagios.OK:
+        print "%s: All %s volumes are under the warning threshold" % (nagios.status_code_to_label(ret), len(results))
+    elif ret == nagios.UNKNOWN:
+        print "%s: Unable to determine usage on %s volumes" % (nagios.status_code_to_label(ret), counts[nagios.UNKNOWN])
+    elif ret == nagios.WARN:
+        print "%s: There are %s volumes over the warning threshold" % (nagios.status_code_to_label(ret), counts[nagios.WARN])
+    else:
+        print "%s: There are %s volumes over the critical threshold and %s volumes over the warning threshold" % (nagios.status_code_to_label(ret), counts[nagios.CRIT], counts[nagios.WARN])
+
+    for pod, mount, disk_usage, inode_usage, status in results:
+        print "%s: %s:%s - bytes used: %s%%, inodes used: %s%%" % (nagios.status_code_to_label(status), pod, mount, disk_usage, inode_usage)
+
+    return ret
 
-def report():
-    raise NotImplementedError("report is not implemented yet")
 
 def main():
-    project = oc("project", "-q")
-    print analize(map(parse_df, exec_in_pods(project, get_running_pod_names(), check_disk_cmd)))
+    parser = generate_parser().parse_args()
+    warn = parser.warn
+    crit = parser.crit
+
+    if crit < warn:
+        raise ValueError("Critical threshold lower than Warning threshold")
+
+    project = openshift.get_project()
+
+    results = []
+
+    for pod, lines in openshift.exec_in_pods(project, openshift.get_running_pod_names(
+            project), check_disk_cmd):
+        for d in analize(pod, parse_df_lines(lines), warn, crit):
```

Comment:
The results of the analize return a lists of tuples that have information about each disk.  Since the data from exec comes in as multiple lines per pod, I am denormalizing the data so that each result has the pod name, the disk name, usage numbers, and the status of its threshold check.  This is just taking all those separate items and adding them to a list which has the data across all pods/volumes so that reporting is simply able to dump each row.